### DISCUSSION TITLE
compiling for "ESP32 dev Module"

### DISCUSSION THREAD

**Discussion by AndiP-IoT** [Egyras/HeishaMon#610] [Q&A]
Hi!
I've tried to compile for the "ESP32 dev module". After setting a partition scheme "Minimal SPIFFS (1,9MB APP with OTA/190kB SPIFFS)" it compiled and flashed. (Arduino GIU 1.8.19)
But it is not running.
I got some messages over serialmonitor an then a reset - over and over again.
`Starting debugging, version: 3.8

--- HEISHAMON ---
starting...
Starting littlefs...
Started littlefs...
Checking littlefs for first boot...
Heishamon boot file exists, normal boot...
Send current wifi info to serial...
Mode: ets Jun  8 2016 00:22:57

rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:4916
load:0x40078000,len:16436
load:0x40080400,len:4
ho 8 tail 4 room 4
load:0x40080404,len:3524
entry 0x400805b8
`
Did I miss something?

Thank you, Andi

**Comment by geduxas** [comment]
Look to GitHub workflow, if your compiler flags are same, it should be working.

**Comment by IgorYbema** [comment]
and make sure you are not using esp32 library 3.1.0 (which is just released). This fails to work properly on the current heishamon code.

**Reply by AndiP-IoT** [reply]
> and make sure you are not using esp32 library 3.1.0 (which is just released). This fails to work properly on the current heishamon code.


I've changed the board definition lib to V 3.0.1-> different message.
`ELF file SHA256: e852e2a7bdbc1fe8

Rebooting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1448
load:0x40078000,len:14844
ho 0 tail 12 room 4
load:0x40080400,len:4
load:0x40080404,len:3356
entry 0x4008059c
Starting debugging, version: 3.8

--- HEISHAMON ---
starting...
Starting littlefs...
Started littlefs...
Checking littlefs for first boot...
Heishamon boot file exists, normal boot...
Send current wifi info to serial...
Mode: Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.

Core  1 register dump:
PC      : 0x4008c5f4  PS      : 0x00060c30  A0      : 0x800fb675  A1      : 0x3ffb20e0  
A2      : 0xbb49e509  A3      : 0xbb49e505  A4      : 0x000000ff  A5      : 0x0000ff00  
A6      : 0x00ff0000  A7      : 0xff000000  A8      : 0x800faab4  A9      : 0x3ffb20b0  
A10     : 0x3ffbe1ac  A11     : 0x3f424dec  A12     : 0x00000006  A13     : 0xffffffff  
A14     : 0x00ff0000  A15     : 0xff000000  SAR     : 0x00000000  EXCCAUSE: 0x0000001c  
EXCVADDR: 0xbb49e509  LBEG    : 0x4008c615  LEND    : 0x4008c625  LCOUNT  : 0xfffffffe  


Backtrace: 0x4008c5f1:0x3ffb20e0 0x400fb672:0x3ffb20f0 0x400fb779:0x3ffb2110 0x400efd67:0x3ffb2130 0x400d6726:0x3ffb2230 0x400fcadf:0x3ffb2290

`
Is the setting for the partition scheme correct? It seems, I need at least 1,9MB APP 
Can I strip something? i.e OTA or NeoPixel, Display ....
I would be happy with only MQTT (hardcoded in source)

> Look to GitHub workflow, if your compiler flags are same, it should be working.


I did. That's how I found the partition scheme. But I'm not used to use a yaml file. Can I use it to configure the arduino IDE?

**Reply by AndiP-IoT** [reply]
it seems, I used the "reply" and quoting the wrong way?

**Comment by IgorYbema** [comment]
Did you also enable USB CDC and PSRAM?
Are you building it for the ESP32-S3 (like on the heishamon large) or for another board/proc?

What is the reason for you to compile it yourself?

**Reply by AndiP-IoT** [reply]
> Did you also enable USB CDC and PSRAM? Are you building it for the ESP32-S3 (like on the heishamon large) or for another board/proc?

I can't find USB CDC in the IDE? - sorry
PSRAM is enabled now (after your post) but not working
`E (415) quad_psram: PSRAM ID read error: 0xffffffff, PSRAM chip not found or not supported`

It's a AZ-delivery ESP32-WROOM-32 chip.
I selected "ESP32 dev Module". Is there a significant difference between ESP32 and ESP32-S3 which touches the program itself? Apart from "native USB", other CPU (compile should solve), bluetooth (I don't need)
I use both:
https://www.az-delivery.de/en/products/esp32-developmentboard
https://www.az-delivery.de/en/products/esp-32-dev-kit-c-v4

> What is the reason for you to compile it yourself?
Well, I would like to play with the source code - to understand - to learn - to see how panasonic does the communication.

Thank you!

**Comment by IgorYbema** [comment]
Just understand that other than the ESP8266 and the ESP32-S3 the code hasn't been tested to work so you might fall into these kind of issue's.

It seems to fail on the wifi printdiaq call as that is right after the "Send current wifi info to serial..." message. So if you really want to learn you might start focussing on this part. But also the code isn't ready for other ESP32 models so it might also just fail on the pinout definitions.

**Reply by AndiP-IoT** [reply]
Thank you!

> Just understand that other than the ESP8266 and the ESP32-S3 the code hasn't been tested to work so you might fall into these kind of issue's.
I see.
I also tried ESP8266 (arduino IDO Board - "NodeMCU 1.0 (ESP-12E Module)" but ran in the "'MEMPOOL_ADDRESS' was not declared in this scope" issue. -> settings like in https://github.com/Egyras/HeishaMon/issues/524 -> it compiled and flashed.
But no wifi hotspot  -> 

OK, I've missed to press RESET 2x fast
It's workin with the ESP8266 AMICA board! - Thank You !!

> It seems to fail on the wifi printdiaq call as that is right after the "Send current wifi info to serial..." message. So if you really want to learn you might start focussing on this part. But also the code isn't ready for other ESP32 models so it might also just fail on the pinout definitions.

Ok, I will see.
I think in the source code is a lot functionality, I don't need. Is there a "naked" heatpump (mqtt) communication protocoll only softwarepart available?


Thank you very much so far!!!!! It is important to be kicked in the right direction.
Regards, Andi

**Comment by IgorYbema** [comment]
You can remove all dallas/onewire, s0, opentherm and gpio code if you want. But those are already disabled per default in the settings so they don't do anything if they are disabled.

**Reply by AndiP-IoT** [reply]
> You can remove all dallas/onewire, s0, opentherm and gpio code if you want. 
OK.
> But those are already disabled per default in the settings so they don't do anything if they are disabled.
I see, thank you.
